app/src/main/java/org/mozilla/reference/browser/BrowserApplication.kt

```diff
@@ -30,6 +30,11 @@ open class BrowserApplication : Application() {
     override fun onCreate() {
         super.onCreate()
 
+        // Set up Glean as early as possible in order to ensure availability to the crash reporting
+        // service.  Glean provides its own assurances that it is running on the main process so it
+        // should be safe calling the following function from another process.
+        setupGlean(this)
```

Comment:
I wonder why we do that in every process now - like the comment says `initialize()` exits early if not on the main process? 🤔

Reply:
Unfortunately, if Glean isn't initialized before the init of the GleanCrashReporterService then there's no way for it to record data in Glean since it relies on being initialized in the main process where Glean is initialized.  I really wish I could have found a better way to record the data without having to use a file to persist the data and then relying on the init to run in the same process as the initialized Glean singleton.

Reply:
In this case, Glean should be a no-op for the purposes of if this is called off of the main process, but we need to initialize the crash reporter on every process.  If only I didn't need Glean to be initialized before the crash reporter (but only when there are crashes persisted that need recorded in Glean).